src/shared/ui/SketchCanvas/LineSketcher.ts

```diff
@@ -85,10 +89,14 @@ class LineSketcher {
   }
 
   public progressLine(
-    path: SkPath,
+    path: SkPath | undefined,
     point: Point,
     straightLine: boolean = false,
   ): void {
+    if (!path) {
+      return;
+    }
+
```

Comment:
Please elaborate on why the `path` variable can be undefined in the `progressLine` and `addPointToPath`

Reply:
@vmkhitaryanscn I see in this case, I would propose creating a separate method, `.cancel()`. This method should clear the current line and cancel the drawing gesture.

Reply:
If it is not possible to cancel the gesture, please add a new method to the LineSketcher class called `canProgressLine` that will be responsible for this type of check. Call it before lineSketcher.progressLine. Also, add the check inside the `shouldCreateNewLine` method